### DISCUSSION TITLE
Is there an easy way to initialize the data prop?

### DISCUSSION THREAD

**Discussion by AFK-Python** [puckeditor/puck#93] [Q&A]
Hi again,

Is there an easy way to initialize the data property that is obtained from `getPage(path);` function and then passed to the `Render` and `Puck` components? Or do I have to build this view the puck editor (ironic that I would prefer the former haha)?

It would be nice to be able to quickly design a basic/default version of the site for users to be able to use and also allow them to add/modify routes from the editor. The latter feature I am asking for seems to be part of the `getPage` function as far as I can tell, but I could be wrong.

Thanks in advance,
Alex

**Comment by chrisvxd** [comment]
I assume you're using the generator here.

If you create a page manually, you should be able to get the page data out of the JSON database (or your own database, if you've switched it out) and pass it in to <Client>:

```tsx
  const defaultData = {}; // Manually created using Puck

  return <Client isEdit={isEdit} path={path} data={data || defaultData} />;
```

Alternatively, you could put a console.log in your `onPublish` callback, can create a page manually with Puck, publish it and copy the output.

Hope that helps!

**Reply by AFK-Python** [reply]
That makes sense. I'll probably go with `console.log` route.

Thanks again!
